### PR TITLE
Liking & Commenting

### PR DESCRIPTION
**Pull request by AbelKel**
**New Features:** 
* Users can now like a college in the details view 
* Users can comment under a college 
* Likes Page shows liked colleges 
* Users can upload a profile picture if they're logged in with Parse [not FB]
**Bugs:**
* Comment is duplicated multiple times in the details view. I stored the comments as values for college names which are the keys in a dictionary. It probably has something to do with that. 
* Profile pictures does not update immediately after uploading. I have to restart the app. 
**For Future PRs:**
* Fixes for the bugs mentioned above
* Have a working version of the first ambiguous problem I am planning to tackle, making asynchronous API calls for filtering

Screen Shot 2022-07-13 at 8 36 38 AM
Screen Shot 2022-07-13 at 8 37 00 AM

### CODE REVIEW

File: Capstone/Like View/LikeCell.h [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+#import <UIKit/UIKit.h>
+#import "College.h"
+NS_ASSUME_NONNULL_BEGIN
+
+@interface LikeCell : UITableViewCell
+@property (strong, nonatomic) College *college;
+@property (weak, nonatomic) IBOutlet UIImageView *likedCollegeImage;
+@property (weak, nonatomic) IBOutlet UILabel *likedCollegeName;
+@property (weak, nonatomic) IBOutlet UILabel *likedCollegeLocation;
+@property (weak, nonatomic) IBOutlet UILabel *likedCollegeDescription;
```
Comment (angelacaq) [inline Capstone/Like View/LikeCell.h:17]: This is likely habit, but let's move this to .m!

File: Capstone/Like View/LikeCell.h [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+//
+//  Created by Abel Kelbessa on 7/12/22.
+//
+
+#import <UIKit/UIKit.h>
+#import "College.h"
+NS_ASSUME_NONNULL_BEGIN
+
+@interface LikeCell : UITableViewCell
+@property (strong, nonatomic) College *college;
```
Comment (angelacaq) [inline Capstone/Like View/LikeCell.h]: Tip! If you have something like this:
```suggestion
@property (strong, readwrite, nonatomic) College *college;
```

In the .m file, you'll get a setter method for free. In that setter method, you can make all the UI updates:
```
- (void)setCollege:(College *)college {
   self.college = college;
   [self buildLikeCell];
}
```

That way, in LikeViewController, you won't need to call `[self buildLikeCell`] — you can just set the college to the cell, and the UI updates will happen for free

File: Capstone/Like View/LikeViewController.m [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+    NSArray *likedCollegeNames;
+    NSArray<College *> *colleges;
+    NSMutableArray *likedCollegesToDisplay;
+}
+
+- (void)viewDidLoad {
+    [super viewDidLoad];
+    self.tableView.delegate = self;
+    self.tableView.dataSource = self;
+    self->likedCollegeNames = [[NSArray alloc] init];
```
Comment (angelacaq) [inline Capstone/Like View/LikeViewController.m:29]:
```suggestion
```

Line isn't necessary if you're replacing this empty array with `current[@"likes"]` below

**Review by angelacaq — APPROVED** [review]
This PR looks great! Great title, branch name, summary, and PR size, which makes this much easier to review :)

As mentioned offline, let's try to find a more efficient method of storing and retrieving likes using Parse in a future PR!

File: Capstone/Like View/LikeCell.m [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+@implementation LikeCell
+
+- (void)awakeFromNib {
+    [super awakeFromNib];
+    // Initialization code
+}
+
+- (void)setSelected:(BOOL)selected animated:(BOOL)animated {
+    [super setSelected:selected animated:animated];
+
```
Comment (skhezri) [inline Capstone/Like View/LikeCell.m]: nit: extra line

File: Capstone/Like View/LikeViewController.m [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+@end
+
+@implementation LikeViewController
+
+- (void)viewDidLoad {
+    [super viewDidLoad];
+}
+
+
+
```
Comment (skhezri) [inline Capstone/Like View/LikeViewController.m]: nit: extra lines

**Review by skhezri — APPROVED** [review]
Great PR Abel!! and thanks for such an in-depth test plan. Next time, try and include a screen recording if the PR includes multiple different functionalities!

**Pull request merged**